packages/server/src/routes/provider/index.svelte

```diff
@@ -1,6 +1,9 @@
 <script>
-  import { goto } from "@sapper/app";
+  import { goto, stores } from "@sapper/app";
   import ResourceSearch from "./../../components/ResourceSearch.svelte";
+
+  const { session } = stores();
+  const isLoggedIn = $session && !!$session.user;
```

Comment:
Does sapper handle session logic for us?

Reply:
yee

Reply:
The sapper middleware has a mechanism to pluck things out of the server-side session and make them available to rendered pages via a Svelte Store.

Doc: https://sapper.svelte.dev/docs#Seeding_session_data
Impl: https://github.com/CodeForBoulder/upswyng/blob/master/packages/server/src/server.js#L125

Reply:
To close the loop ->

There's session middleware on the server to store sessions in mongodb (we could've used redis for this too). The Oauth/Grant middleware sets session data and handles the cookie and such, so that the request on the server gets populated with session data (currently just the logged in user). Then the sapper middleware pulls it out so you can access it on svelte pages.

Reply:
Actually the cookie part is the session middleware

Reply:
That is sweeeeeeeet stuff